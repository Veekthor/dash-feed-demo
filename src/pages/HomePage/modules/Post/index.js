import { PostContainer, UserInfo } from "./index.styled";
import ProfileImg from "../../../../assets/user.png";

const Post = ({ post, isShared, isChild, children }) => {
  const { user } = post;
  const testId = !isChild ? "post-container" : "";
  const firstName = user.name.split(" ")[0];
  const avatarImg = user.profile_picture || ProfileImg;
  return (
    <PostContainer isChild={isChild} data-testid={testId} className="card">
      <UserInfo>
        <img className="avatar" src={avatarImg} alt={firstName} />
        <div>
          <p className="user_name">{user.name}</p>
          <p className="course_of_study">{user.course_of_study}</p>
        </div>
      </UserInfo>
      {isShared ? (
        children
      ) : (
        <>
          <div>
            {post.image && <img src={post.image} alt="" />}
            <p>{post.body}</p>
          </div>
          <div>
            <span>{post.likes} Likes </span>
            <span>{post.comments} Comments </span>
            <span>{post.shares} Shares</span>
          </div>
        </>
      )}
    </PostContainer>
  );
};

export default Post;
