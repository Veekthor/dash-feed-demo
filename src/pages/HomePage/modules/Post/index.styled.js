import styled from "styled-components";

export const PostContainer = styled.div`
  ${({ isChild }) =>
    isChild &&
    `
    margin-left: 30px !important;
    box-shadow: none !important;
  `}
`;
export const UserInfo = styled.div`
  display: flex;
  gap: 15px;

  .user_name {
    font-weight: bold;
  }

  .course_of_study {
    font-size: 11px;
  }
`;
