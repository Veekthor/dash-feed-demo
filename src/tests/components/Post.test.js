import renderer from "react-test-renderer";
import Post from "../../pages/HomePage/modules/Post";
import postsResponse from "../fixtures/postsResponse";
describe("Post Component page", () => {
  it("should match snapshot", () => {
    const component = renderer.create(<Post post={postsResponse.posts[0]} />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("Shared posts should match snapshot", () => {
    const post = postsResponse.posts[4];
    const component = renderer.create(
      <Post isShared post={{ user: post.shared_by }}>
        <Post isChild post={post} />
      </Post>
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
