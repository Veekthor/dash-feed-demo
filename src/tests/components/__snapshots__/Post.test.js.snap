// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Post Component page Shared posts should match snapshot 1`] = `
<div
  className="sc-aXZVg fRzBGV card"
  data-testid="post-container"
>
  <div
    className="sc-gEvEer jWtOBt"
  >
    <img
      alt="Urooj"
      className="avatar"
      src="https://my-campus-bucket-staging.s3.amazonaws.com/media/profile_picture/23/01/19/c675af77-eae6-4fa6-b6d5-746d732e6e87.jpeg"
    />
    <div>
      <p
        className="user_name"
      >
        Urooj Khurshid
      </p>
      <p
        className="course_of_study"
      />
    </div>
  </div>
  <div
    className="sc-aXZVg kUeanE card"
    data-testid=""
  >
    <div
      className="sc-gEvEer jWtOBt"
    >
      <img
        alt="Steven"
        className="avatar"
        src="user.png"
      />
      <div>
        <p
          className="user_name"
        >
          Steven David
        </p>
        <p
          className="course_of_study"
        >
          accelerated professional science masters 
        </p>
      </div>
    </div>
    <div>
      <p>
        New Space. Nice ...
      </p>
    </div>
    <div>
      <span>
        0
         Likes 
      </span>
      <span>
        0
         Comments 
      </span>
      <span>
        1
         Shares
      </span>
    </div>
  </div>
</div>
`;

exports[`Post Component page should match snapshot 1`] = `
<div
  className="sc-aXZVg fRzBGV card"
  data-testid="post-container"
>
  <div
    className="sc-gEvEer jWtOBt"
  >
    <img
      alt="Steven"
      className="avatar"
      src="user.png"
    />
    <div>
      <p
        className="user_name"
      >
        Steven David
      </p>
      <p
        className="course_of_study"
      >
        accelerated professional science masters 
      </p>
    </div>
  </div>
  <div>
    <p>
      New Post Like Pls
    </p>
  </div>
  <div>
    <span>
      2
       Likes 
    </span>
    <span>
      0
       Comments 
    </span>
    <span>
      0
       Shares
    </span>
  </div>
</div>
`;
